### ISSUE TITLE
General question: cooperation with 'musicexamples'

### ISSUE THREAD

**Issue by uliska** [jperon/lyluatex#26]
Having finally looked into `lyluatex` I think (even more than initially) that `lyluatex` and my [`musicexamples`](https://github.com/uliska/musicexamples) package would go together very well, and we should discuss how that could work best.

`lyluatex` manages LilyPond source code and compilation with suitable dimensions but “only“ integrates them in the final document as images included with `\includegraphics` or `\includepdf`.
`musicexamples` provides commands and environments to properly integrate music examples into LaTeX documents but is only able to deal with PDFs generated separately. So actually there is absolutely no overlap currently, and each package implements what the other misses, which makes them ideal companions.

Among the elements `musicexamples` adds to the mix are

* floating and non-floating environments for music examples  
  (=> allows the use of captions and labels)
* Many different environment styles (wrapped, full-line, sideways)
* Support for full-page scores where captions are moved to the header/footer
* Support for multi-page scores where the current text is not interrupted (i.e. text flows until the end of the page and continues after the multipage score)
* Support for multi-page scores that start on the next odd or even page (so there may be 1 1/2 pages of text until the score starts) to enforce page turns. (This isn't fully worked through and will have to be adjusted to a lyluatex use case)
* Print separate list of musicexamples

---

What options do we have?

1. integrate `musicexamples` into `lyluatex`
2. integrate `lyluatex` into `musicexamples`
3. Don't integrate them but make sure they work together seamlessly (and bring `musicexamples` to a state where it can be published on CTAN)
4. as 3. but "use" `musicexamples` from within `lyluatex`

I vote for 3. or 4. I would not strictly bind the two together because `musicexamples` should work with *any* TeX engine, and I don't think it's a good idea to add luatex as a requirement (although I personally use luatex exclusively).
OTOH it would be good to have the functionality available in one single package. Of course it would be possible to wrap the lyluatex functions to only be available when run on luatex, but probably that's overly complicated.

One alternative could be to reimplement lyluatex using `pythontex`. While this would *still* add a heavy dependency - Python - it would still allow any TeX engine to be used.
Or we could try to reimplement everything in pure (La)TeX, based on the code shared by Knut Petersen (although I'm not sure to what extent this code can be made as maintainable as lyluatex's current one - at least *I* will not be of much help here).

---

So my suggestion is to

* Continue developing `lyluatex` as it is, aiming for more versatility and configurability
* Bring `musicexamples` up to par (and publish it), review the implementation, and remove all the attempts at working on the LilyPond compilation part
* Update the documentation of both packages showing how they can be used together
* Wait and see if there comes an option to make lyluatex less dependent on luatex

**Comment by uliska** [comment]
Showcase document for `musicexamples`: [xmp-Example.pdf](https://github.com/jperon/lyluatex/files/1632067/xmp-Example.pdf)

**Comment by jperon** [comment]
I really think option 3 is the best. Perhaps we could make a mix with 4 by providing an option, so that one could integrate both packages with one `\includepackage` ; but IMHO, this would be a possible cause of difficult to understand problems for users.

**Comment by uliska** [comment]
OK, I agree. So let's go for my third suggestion and create two complementary packages that are designed and documented to work together.

I have started reviewing the documentation of `musicexamples` by throwing out all references to the Python attempts. At the same time I could close five out of 10 issues because `lyluatex` exactly provides the functionality I was wishing for back in 2014 when I stopped working on it.
